docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "        Path (as str, as expected by `apache_beam`) to the reference dataset file\n",
+    "        \n",
+    "    Returns\n",
+    "    -------\n",
+    "    beam.PCollection: The (parallel) collection of `MovieTitle` instances, ready for further processing.\n",
+    "    \"\"\"\n",
+    "    return (\n",
+    "        p\n",
+    "        | \"Read MOvieTitle from Data File\" >> beam.io.ReadFromText(data_file)\n",
+    "        | \"Parse into MovieTitle objects\" >> beam.ParDo(ParseMovieTitles()).with_output_types(MovieTitle)\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Run Pipeline**\n",
+    "\n",
+    "The last component of our framework will be devoted the actual _execution_ of the pipeline. In particular, we will define the _core_ `run_pipeline` function which will accept the following three parameters:\n",
+    "\n",
+    "- `pipeline_fn`: the _pluggable_ `PipelineModule` to execute;\n",
+    "\n",
+    "\n",
+    "- `pcollection_fn`: the selected callable `PCollectionFun` to call to gather the target data objects `PCollection` (either `netflix_movie_views_collection` or `netflix_movie_title_collection`)\n",
+    "\n",
+    "\n",
+    "- `running_mode`: execution mode to be passed on as `PipelineOption`. Accepted values are (I) `multi_processing` (default); (II) `multi_threading`; (III) `in_memory`. More information on the pipeline options are available in the official [documentation](https://beam.apache.org/documentation/patterns/pipeline-options/);\n",
+    "\n",
+    "\n",
+    "\n",
+    "- `data_file`: string matching the path to the input `data_file` (to be passed to the `netflix_movie_views_collection` function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# A PipelineModule is defined as a Callable that accepts any parameter\n",
+    "# and returns a beam.PCollection` instance\n",
+    "\n",
+    "from typing import Callable\n",
+    "\n",
+    "PipelineModule = Callable[[beam.PCollection], beam.PCollection]\n",
+    "PCollectionFun = Callable[[beam.Pipeline, str], beam.PCollection]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title **Check** Python multiprocessing settings { display-mode: \"form\" }\n",
+    "import multiprocessing as mp\n",
+    "\n",
+    "print(\"Python Multiprocessing Start method: \", mp.get_start_method())\n",
+    "print(\"Multiprocessing available Cores: \", mp.cpu_count())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "from apache_beam.runners import DirectRunner\n",
+    "from apache_beam.options.pipeline_options import PipelineOptions\n",
+    "\n",
+    "def run_pipeline(pipeline_fn: PipelineModule, \n",
```

Comment:
could you please add
#@title ....
and hide the code by default (it might be too much information for the user) ?